src/Compilers/VisualBasic/Portable/Lowering/Instrumentation/DynamicAnalysisInjector.vb

```diff
@@ -0,0 +1,303 @@
+            Select Case statement.Kind
+                Case BoundKind.IfStatement
+                    Return DirectCast(statement, BoundIfStatement).Condition.Syntax
+                Case BoundKind.WhileStatement
+                    Return DirectCast(statement, BoundWhileStatement).Condition.Syntax
+                Case BoundKind.ForToStatement
+                    Return DirectCast(statement, BoundForToStatement).InitialValue.Syntax
+                Case BoundKind.ForEachStatement
+                    Return DirectCast(statement, BoundForEachStatement).Collection.Syntax
+                Case BoundKind.DoLoopStatement
+                    Return DirectCast(statement, BoundDoLoopStatement).ConditionOpt.Syntax
+                Case BoundKind.UsingStatement
+                    Dim usingStatement As BoundUsingStatement = DirectCast(statement, BoundUsingStatement)
+                    Return If(usingStatement.ResourceExpressionOpt IsNot Nothing, DirectCast(usingStatement.ResourceExpressionOpt, BoundNode), usingStatement).Syntax
+                Case BoundKind.SyncLockStatement
+                    Return DirectCast(statement, BoundSyncLockStatement).LockExpression.Syntax
+                Case BoundKind.SelectStatement
+                    Return DirectCast(statement, BoundSelectStatement).ExpressionStatement.Expression.Syntax
+                Case BoundKind.LocalDeclaration
+                    Dim initializer As BoundExpression = DirectCast(statement, BoundLocalDeclaration).InitializerOpt
+                    If initializer IsNot Nothing Then
+                        Return initializer.Syntax
+                    End If
+            End Select
+
+            Return statement.Syntax
+        End Function
+
+        Private Shared Function MethodHasExplicitBlock(method As MethodSymbol) As Boolean
+            Return TypeOf TryCast(method.OriginalDefinition, SourceMethodSymbol)?.Syntax Is MethodBlockBaseSyntax
+        End Function
+
+        Private Shared Function GetCreatePayload(compilation As VisualBasicCompilation, syntax As VisualBasicSyntaxNode, diagnostics As DiagnosticBag) As MethodSymbol
+            Return DirectCast(Binder.GetWellKnownTypeMember(compilation, WellKnownMember.Microsoft_CodeAnalysis_Runtime_Instrumentation__CreatePayload, syntax, diagnostics), MethodSymbol)
+        End Function
+
+        Private Shared Function MethodDeclarationIfAvailable(body As VisualBasicSyntaxNode) As VisualBasicSyntaxNode
+            Dim parent As VisualBasicSyntaxNode = body.Parent
+            If parent IsNot Nothing Then
+                Select Case (parent.Kind())
+                    Case SyntaxKind.FunctionBlock
+                    Case SyntaxKind.SubBlock
+                    Case SyntaxKind.PropertyBlock
+                    Case SyntaxKind.GetAccessorBlock
+                    Case SyntaxKind.SetAccessorBlock
+                        Return parent
```

Comment:
This entire method turns out to be a placebo and can be removed. The body syntax node is already the desired one, and so all calls were taking the default path.